packages/keys/src/public-key.ts

```diff
@@ -0,0 +1,19 @@
+import { assertKeyExporterIsAvailable } from '@solana/assertions';
+
+export async function getPublicKeyFromPrivateKey(privateKey: CryptoKey, extractable?: boolean): Promise<CryptoKey> {
+    assertKeyExporterIsAvailable();
+
+    // Export private key.
+    const jwk = await crypto.subtle.exportKey('jwk', privateKey);
+
+    // Remove private data from JWK.
+    delete jwk.d;
+    delete jwk.dp;
+    delete jwk.dq;
+    delete jwk.q;
+    delete jwk.qi;
+    jwk.key_ops = ['verify'];
+
+    // Import public key.
+    return await crypto.subtle.importKey('jwk', jwk, 'Ed25519', extractable ?? false, ['verify']);
```

Comment:
I think it's preferable to build this up according to what the importer needs, rather than to delete things. What if, in the future, the export API starts returning extra crap that we don't have a `delete` statement for?
```suggestion
    // Import public key.
    return await crypto.subtle.importKey(
        'jwk',
        {
            crv /* curve */: 'Ed25519',
            ext /* extractable */: extractable ?? false,
            key_ops /* key operations */: ['verify'],
            kty /* key type */: 'OKP' /* octet key pair */,
            x /* public key x-coordinate */: jwk.x,
        },
        'Ed25519',
        extractable ?? false,
        ['verify'],
    );
```